Basic-Car-Maintenance/Shared/Settings/Views/AddVehicleView.swift

```diff
@@ -42,7 +41,11 @@ struct AddVehicleView: View {
                 ToolbarItem {
                     Button {
                         let vehicle = Vehicle(name: name, make: make, model: model)
-                        addTapped(vehicle)
+                        
+                        if !name.isEmpty && !make.isEmpty && !model.isEmpty {
+                            addTapped(vehicle)
+                        }
+                        
```

Comment:
A better way to do this is make this a computed property `!name.isEmpty && !make.isEmpty && !model.isEmpty` and disable the Add button until those parameters are met

Otherwise the user will be tapping Add and will be confused why nothing is happening

Reply:
Made the requested changes.